### PR TITLE
fix Issue 14157. Part 2

### PR DESCRIPTION
**Pull request by 9il**
See also #1162

### CODE REVIEW

**Comment by schveiguy** [comment]
I don't think this is right. The docs I looked at showed all were called `fabs`, I'd be surprised if this passes.

**Comment by 9il** [comment]
my bad

**Comment by schveiguy** [comment]
BTW, how did the original even pass? Are there any unit tests that use fabs?

**Comment by 9il** [comment]
No tests: all `std.*` use `std.math.fabs`.

**Comment by 9il** [comment]
fixed

**Comment by schveiguy** [comment]
> No tests

Then how do you know it links? ;)

I don't know that we need to start adding unit tests for stdc bindings, but I think at least we should try linking locally on a win64 box, which I assume you have, right?

**Comment by 9il** [comment]
No, I use OS X.
This PR https://github.com/D-Programming-Language/phobos/pull/2513 can check `fabs`.

**Comment by schveiguy** [comment]
OK, I have a Win64 VM. I will try this when I get a chance.

**Comment by 9il** [comment]
Thanks!

**Comment by schveiguy** [comment]
Closing this, as `long double` is identical to `double` on windows. See https://msdn.microsoft.com/en-us/library/9cx8xs15.aspx

**Comment by schveiguy** [comment]
Reopening to allow removal of real overload.

File: src/core/stdc/tgmath.d [hunk 1]
```diff
@@ -869,10 +869,23 @@ else
...
-    ///
-    alias core.stdc.math.fabsl         fabs;
+    version( CRuntime_Microsoft )
+    {
+        extern(C++)
+        {
+            ///
+            float   fabs(float x);
+            ///
+            real    fabs(real x);
```
Comment (9il) [inline src/core/stdc/tgmath.d]: Should I remove `C++` prototypes in `core.stdc.tgmath` ?

Comment (schveiguy) [inline src/core/stdc/tgmath.d]: Makes sense. With the double being the c version, it covers both float and double

**Comment by schveiguy** [comment]
Auto-merge toggled on

**Pull request merged**